src/root/components/Alert/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import {
+  AiOutlineInfoCircle,
+  AiOutlineWarning,
+  AiOutlineCheckCircle,
+  AiOutlineClose,
+  AiOutlineQuestionCircle,
+} from 'react-icons/ai';
+
+import { AlertVariant } from '#components/AlertContext';
+import ElementFragments from '#components/ElementFragments';
+import RawButton from '#components/RawButton';
+
+import styles from './styles.module.scss';
+
+export interface Props<N> {
+  name: N;
+  className?: string;
+  variant?: AlertVariant;
+  children: React.ReactNode;
+  nonDismissable?: boolean;
+  onCloseButtonClick?: (name: N) => void;
+}
+
+const alertVariantToClassNameMap: {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  [key in AlertVariant]: string;
+} = {
+  success: styles.success,
+  warning: styles.warning,
+  danger: styles.danger,
+  info: styles.info,
+};
+
+function Alert<N extends string>(props: Props<N>) {
+  const {
+    name,
+    className,
+    variant = 'info',
+    children,
+    onCloseButtonClick,
+    nonDismissable,
+  } = props;
+
+  const icon: {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    [key in AlertVariant]: React.ReactNode;
+  } = {
+    success: <AiOutlineCheckCircle className={styles.icon} />,
+    danger: <AiOutlineWarning className={styles.icon} />,
+    info: <AiOutlineInfoCircle className={styles.icon} />,
+    warning: <AiOutlineQuestionCircle className={styles.icon} />,
+  };
```

Comment:
Let' move this outside the render function. This doesn't need to be calculated everytime.